specification/extendedlocation/resource-manager/Microsoft.ExtendedLocation/preview/2021-08-31-preview/customlocations.json

```diff
@@ -0,0 +1,1196 @@
+            "type": "string"
+          }
+        }
+      }
+    },
+    "resourceSyncRule": {
+      "description": "Resource Sync Rules definition.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/resourceSyncRuleProperties",
+          "description": "The set of properties specific to a Resource Sync Rule",
+          "x-ms-client-flatten": true
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "description": "Metadata pertaining to creation and last modification of the resource",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/TrackedResource"
+        }
+      ]
+    },
+    "resourceSyncRuleListResult": {
+      "description": "The List Resource Sync Rules operation response.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The URL to use for getting the next set of results.",
+          "type": "string",
+          "readOnly": true
+        },
+        "value": {
+          "description": "The list of Resource Sync Rules.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/resourceSyncRule"
+          }
+        }
+      }
+    },
+    "resourceSyncRuleProperties": {
+      "description": "Properties for a resource sync rule.",
+      "type": "object",
+      "properties": {
+        "priority": {
+          "description": "Priority represents a priority of the Resource Sync Rule",
+          "type": "integer",
+          "format": "int32"
+        },
+        "provisioningState": {
+          "description": "Provisioning State for the Resource Sync Rule.",
+          "type": "string",
+          "readOnly": true
+        },
+        "selector": {
+          "description": "A label selector is a label query over a set of resources. The result of match labels and match expressions should both be satisfied.",
+          "type": "object",
+          "properties": {
+            "matchExpressions": {
+              "$ref": "#/definitions/matchExpressionsList",
+              "description": "MatchExpressions contain values, a key, and an operator that relate the key and values.",
+              "x-ms-client-flatten": true
+            },
+            "matchLabels": {
+              "description": "MatchLabels is a map of {key,value} pairs",
+              "type": "object",
+              "additionalProperties": {
+                "type": "string"
+              }
+            }
+          }
+        },
+        "targetResourceGroup": {
+          "description": "Resource Group, all resources will be projected to. The Custom Locations RP will perform a checkAccess API for write permissions on the resource group.",
```

Comment:
what explicit permissions will be checked?  How do you know what resource types amy end up getting "matched" to this rule to be able to check all appropriate permissions?

Reply:
Updated the description in the swagger with more details.

Reply:
"write" permissions doesn't mean anything by itself.  What specific actions will you be validating?  What resource types are being created in the target resource group and where does their content come from?

Reply:
some other questions

1. Does the resource group have to already exist?
2. What will happen if the resource group is deleted after the sync rule is created?  How does a user discover the failures?
3. How do you surface failures due to locked resource groups, denys from Azure Policy, or similar during a resource sync?
4. Why isn't this using a managed identity to interact with the target resource group?  If I coerce someone into giving me permissions to their RG then set up a sync rule while I have access how will they ever clean up the sync rule after they've removed my erroneous access?

Reply:
> "write" permissions doesn't mean anything by itself. What specific actions will you be validating? What resource types are being created in the target resource group and where does their content come from?

For the "write" permission, we will validate that the user can create/update the target resource group.
The resource types being projected are resource types enabled and supported in the parent custom location resource.
The content is from off azure resource metadata.

Reply:
> some other questions
> 
> 1. Does the resource group have to already exist?
> 2. What will happen if the resource group is deleted after the sync rule is created?  How does a user discover the failures?
> 3. How do you surface failures due to locked resource groups, denys from Azure Policy, or similar during a resource sync?
> 4. Why isn't this using a managed identity to interact with the target resource group?  If I coerce someone into giving me permissions to their RG then set up a sync rule while I have access how will they ever clean up the sync rule after they've removed my erroneous access?

1
Q: Does the resource group have to already exist?
A: Yes currently the RG would have to exist.

2
Q: What will happen if the resource group is deleted after the sync rule is created? How does a user discover the failures?
A: These resources currently exist off Azure. The end user would be able to see the failure off Azure (failure and failure reason will be annotated back to the off Azure resources). There is some work that we may need to do to bubble up events to customers on Azure side by exposing the synchronization events. We will collect customer's feedback in the private preview and add this in next version.

3
Q: How do you surface failures due to locked resource groups, denys from Azure Policy, or similar during a resource sync?
A: Similar to previous questions. The customer could see the failure off Azure, but we probably need a way to bubble them onto Azure.

4
Q: Why isn't this using a managed identity to interact with the target resource group? If I coerce someone into giving me permissions to their RG then set up a sync rule while I have access how will they ever clean up the sync rule after they've removed my erroneous access?
A:
We chatted with the AAD team on using MSI vs just creating the resources; I think this was a grey area, because the source of truth for these resources exist off Azure. We are just projecting that source of truth into a RG. I think some of the points that you are bringing up are good though. We will collect customer's feedback in the private preview and improve in the next version.